prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/BedrockLLMClient.kt

```
@@ -0,0 +1,292 @@
+                            val chunkBytes = event.asChunk().bytes
+                            if (chunkBytes != null) {
+                                val chunkJsonString = chunkBytes.decodeToString()
+                                send(chunkJsonString)
+                                logger.trace { "Bedrock Stream Chunk for model ${model.id}: $chunkJsonString" }
+                            } else {
+                                logger.warn { "Received null chunk bytes in stream for model ${model.id}" }
+                            }
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                logger.error(e) { "Error in Bedrock streaming for model ${model.id}" }
+                close(e)
+            }
+        }.map { chunkJsonString ->
+            try {
+                if (chunkJsonString.isBlank()) return@map ""
+
+                when (modelFamily) {
+                    is BedrockModelFamilies.AI21Jamba -> BedrockAI21JambaSerialization.parseJambaStreamChunk(chunkJsonString)
+                    is BedrockModelFamilies.AmazonNova -> {
+                        if (model.id.startsWith("amazon.nova")) {
+                            BedrockAmazonNovaSerialization.parseNovaStreamChunk(chunkJsonString)
+                        } else {
+                            // Amazon Titan
+                            throw IllegalArgumentException("Model ${model.id} not supported in koog")
+                        }
+                    }
+                    is BedrockModelFamilies.AnthropicClaude -> BedrockAnthropicClaudeSerialization.parseAnthropicStreamChunk(chunkJsonString)
+                    is BedrockModelFamilies.Meta -> BedrockMetaLlamaSerialization.parseLlamaStreamChunk(chunkJsonString)
+                }
+            } catch (e: Exception) {
+                logger.warn(e) { "Failed to parse Bedrock stream chunk: $chunkJsonString" }
+                ""
```

Let's rethrow an exception explicitly here instead of sliently swallowing it